room/staff.py

```diff
@@ -188,15 +176,9 @@ def get(self):
     entries = list(query)
     logging.info('loaded %d sites for %s', len(entries), user.program_selected)
     sitecaptains_by_site = {}
-    # TODO: this is fetching too many - we only need those for the current
-    # program.  use ancestor?
-    for sc in ndb_models.SiteCaptain.query():
-      sitecaptains_by_site.setdefault(sc.site.integer_id(), []).append(sc)
-    logging.info('loaded sitecaptains')
-    for s in entries:
-      k = s.key.integer_id()
-      if k in sitecaptains_by_site:
-        s.sitecaptains = sitecaptains_by_site[k]
+    for site in entries:
+      site.sitecaptains = list(site.sitecaptain_set)
```

Comment:
There's a query under this sitecaptain_set property, hence the need to do all this dancing around in the first place. 
https://github.com/babybunny/rebuildingtogethercaptain/blob/93f90e72b723660d2f0f7d91fdf32b70aaae283d/room/ndb_models.py#L435